bolt/lib/Profile/StaleProfileMatching.cpp

```diff
@@ -394,7 +400,7 @@ createFlowFunction(const BinaryFunction::BasicBlockOrderType &BlockOrder) {
 void matchWeightsByHashes(BinaryContext &BC,
                           const BinaryFunction::BasicBlockOrderType &BlockOrder,
                           const yaml::bolt::BinaryFunctionProfile &YamlBF,
-                          FlowFunction &Func) {
+                          FlowFunction &Func, uint64_t &MatchedBlocksCount) {
```

Comment:
Why not return MatchedBlocksCount instead?

Reply:
In the case that we want to build upon our profile discrepancy heuristic, I think it makes sense to pass in a reference to "something" (variable, struct) that maintains information related to the heuristic, similarly how to the flow function is treated by this function. But I don't have strong feelings about this - if you do I'm happy to change it.

Reply:
I think it makes sense to return the stats as a result of matching. We can extend the return type into struct in the future if the need arises.

Reply:
Sounds good